shopfloor/services/checkout.py

```diff
@@ -533,15 +536,47 @@ def _select_lines_from_packaging(self, picking, selection_lines, packaging, **kw
         )
 
     def _select_lines_from_lot(self, picking, selection_lines, lot, **kw):
-        lines = selection_lines.filtered(lambda l: l.lot_id == lot)
+        message = None
+        lines = self._picking_lines_by_lot(picking, selection_lines, lot)
         if not lines:
-            return self._response_for_select_line(
-                picking,
-                message={
-                    "message_type": "error",
-                    "body": _("Lot is not in the current transfer."),
-                },
+            if not kw.get("confirm_lot"):
+                lines_same_product = selection_lines.filtered(
+                    lambda l: (
+                        l.product_id == lot.product_id
+                        # We cannot change a lot on a move having ancestors.
+                        # That would mean we already picked up the wrong lot
+                        # on the previous move(s) and Odoo already restricts
+                        # the reservation based on the previous move(s).
+                        and not l.move_id.move_orig_ids
+                    )
+                )
```

Comment:
This deserves to be put out of the scenario. This is valid for all scenario

Reply:
Indeed, this deserves a test too.